azurerm/internal/services/securitycenter/resource_arm_security_center_subscription_pricing.go

```diff
@@ -86,15 +124,17 @@ func resourceArmSecurityCenterSubscriptionPricingRead(d *schema.ResourceData, me
 	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	resp, err := client.GetSubscriptionPricing(ctx, securityCenterSubscriptionPricingName)
+	resource_type := d.Get("resource_type").(string)
```

Comment:
If the above ID type is used, this can be taken from the that ID, rather than the config to ensure we're reading from the API and not config.

Reply:
I'm not sure if I understand; something along the lines of:

```
id := d.GetId()
scPricing := parseId(id)
resp, err := client.Get(ctx, scPricing.resourceType)
```
?

Reply:
something like:

```
id, err := parse.SecurityPricingID(d.Id())
resp, err := client.Get(ctx, id.resourceType)
```

There's examples in most of the resources of this pattern.  Does that help?